src/main/java/uk/gov/companieshouse/overseasentitiesapi/interceptor/FilingInterceptor.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.companieshouse.overseasentitiesapi.interceptor;
+
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+import uk.gov.companieshouse.api.model.transaction.Transaction;
+import uk.gov.companieshouse.overseasentitiesapi.utils.ApiLogger;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.HashMap;
+import java.util.Objects;
+
+import static uk.gov.companieshouse.overseasentitiesapi.utils.Constants.ERIC_REQUEST_ID_KEY;
+import static uk.gov.companieshouse.overseasentitiesapi.utils.Constants.TRANSACTION_ID_KEY;
+import static uk.gov.companieshouse.overseasentitiesapi.utils.Constants.TRANSACTION_KEY;
+
+import static uk.gov.companieshouse.api.model.transaction.TransactionStatus.CLOSED;
+
+@Component
+public class FilingInterceptor implements HandlerInterceptor {
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
```

Comment:
intellij highlights the last 2 method params as needing org.springframework.lang.NonNull annotations. I've added them on the transactionInterceptor, looks like the interface method we override has them.